src/main/java/com/google/netpcapanalysis/dao/PCAPDaoImpl.java

```diff
@@ -157,4 +116,78 @@ public void setMaliciousIPObjects(MaliciousPacket data) {
     datastore.put(Entity);
   }
 
+  private boolean searchFileAttribute(String searchEntity)
+  {
+    FileAttribute temp = new FileAttribute();;
+    Filter propertyFilter = 
+    new FilterPredicate("PCAP_Entity", FilterOperator.EQUAL, searchEntity);
+    Query q = new Query(fileEntity).setFilter(propertyFilter);
+    PreparedQuery pq = datastore.prepare(q);
+
+    List<Entity> result = pq.asList(FetchOptions.Builder.withLimit(1));
+
+    if(result.size() > 0){
+      return true;
+    }
+    else{
+      return false;
+    }
+  }
+
+  private Entity getEntityAttribute(String searchEntity)
+  {
+    FileAttribute temp = new FileAttribute();;
+    Filter propertyFilter = 
+    new FilterPredicate("PCAP_Entity", FilterOperator.EQUAL, searchEntity);
+    Query q = new Query(fileEntity).setFilter(propertyFilter);
+    PreparedQuery pq = datastore.prepare(q);
+
+    List<Entity> result = pq.asList(FetchOptions.Builder.withLimit(1));
+
+    return result.get(0);
+  }
+
+  public FileAttribute getFileAttribute(String searchEntity)
+  {
+    FileAttribute temp = new FileAttribute();;
+    Filter propertyFilter = 
+    new FilterPredicate("PCAP_Entity", FilterOperator.EQUAL, searchEntity);
+    Query q = new Query(fileEntity).setFilter(propertyFilter);
+    PreparedQuery pq = datastore.prepare(q);
+
+    List<Entity> result = pq.asList(FetchOptions.Builder.withLimit(1));
+
+    if(result.size() > 0){
+      String pcapEntity = (String) result.get(0).getProperty("PCAP_Entity");
+      String fileName = (String) result.get(0).getProperty("File_Name");
+      String myIP = (String) result.get(0).getProperty("My_IP");
+      Date uploadDate = (Date) result.get(0).getProperty("Upload_Date");
+       temp = new FileAttribute(pcapEntity, fileName, myIP, uploadDate);
+    }
+    
+   return temp;
+  }
+
+  public void setFileAttribute(FileAttribute data){
+    
+    if(!searchFileAttribute(data.pcapEntity))
```

Comment:
Instead of calling the search first then getEntity combine the two calls into one Datastore  lookup. E.g. Just call the `prevEntity = getFileAttrEntity` and if prevEntity is null then do this.